vllm/engine/protocol.py

```diff
@@ -0,0 +1,84 @@
+from typing import (AsyncIterator, List, Mapping, Optional, Protocol,
+                    runtime_checkable)
+
+from transformers import PreTrainedTokenizer
+
+from vllm.config import DecodingConfig, ModelConfig
+from vllm.core.scheduler import SchedulerOutputs
+from vllm.inputs.data import PromptInputs
+from vllm.lora.request import LoRARequest
+from vllm.outputs import EmbeddingRequestOutput, RequestOutput
+from vllm.pooling_params import PoolingParams
+from vllm.prompt_adapter.request import PromptAdapterRequest
+from vllm.sampling_params import SamplingParams
+from vllm.sequence import SamplerOutput
+
+
+@runtime_checkable
+class VLLMBackend(Protocol):
```

Comment:
Naming wise, I think this the protocol for `AsyncEngineClient` or `AsyncEngine`. Also I prefer no vLLM in the class name. The entire package is named under vLLM. Backend is ambiguous because we already use it for hardware backends, and attention backends.

Reply:
Yeah `AsyncEngine` is pretty spot on

Reply:
I think we should call it a client

Reply:
I don't think we should call it a client, this is the name of the interface that both the `AsyncLLMEngine` itself and the client to the backend process running the `AsyncLLMEngine` implement.

I'd be up for renaming like `RPCClient` -> `AsyncLLMEngineClient` and `RPCServer` -> `AsyncLLMEngineServer` if we wanna be consistent with these names